conversation_event.go

```diff
@@ -0,0 +1,37 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+type EventParams struct {
+	// Type identifies the type of event (start-typing, stop-typing, joined, left)
+	Type string `json:"name"`
+
+	// ParticipantID identifies the message sender.
+	ParticipantID string `json:"participant_id"`
+
+	// ParticipantType identifies the type of participant who sent this message.
+	ParticipantType ParticipantType `json:"participant_type"`
+
+	// Timestamp optionally defines the time when the conversation was assigned.
+	// If not given, this will default to the current time.
+	Timestamp *time.Time `json:"timestamp,omitempty"`
+}
+
+// AssignConversation assigns a conversation to a participant.
+func (c *Client) AddConversationEvent(ctx context.Context, conversationID string, p *EventParams) error {
+	rsp, err := c.makeRequest(ctx, http.MethodPut, fmt.Sprintf("/conversations/%s/event", conversationID), p)
```

Comment:
Let's keep this RESTful 😴 

```suggestion
	rsp, err := c.makeRequest(ctx, http.MethodPost, fmt.Sprintf("/conversations/%s/events", conversationID), p)
```